src/Evolve/Dialect/Cassandra/CassandraCluster.cs

```diff
@@ -39,10 +39,30 @@ protected override void InternalChangeSchema(string toSchemaName)
             new CassandraMetadataTable(schema, tableName, this);
 
         /// <summary>
-        ///     Returns always true, because the lock is granted at table level.
+        ///     Will chekc for a predefined keyspace and table to see if there is a lock.
+        ///     Otherwise, always returns true, because the lock is granted at table level.
         ///     <see cref="CassandraMetadataTable.TryLock"/>
         /// </summary>
-        public override bool TryAcquireApplicationLock() => true;
+        public override bool TryAcquireApplicationLock()
+        {
+            const string ClusterLockKeyspaceName = "cluster_lock";
+            const string ClusterLockTableName = "lock";
+
+            try
+            {
+                return WrappedConnection.QueryForLong($"select count(locked) from {ClusterLockKeyspaceName}.{ClusterLockTableName}") == 0;
+            }
+            catch (EvolveException ex)
+                when (ex?.InnerException.GetType().ToString() == "Cassandra.InvalidQueryException")
+            {
+                //These error messages are very specific to Cassandra's drivers and could change
+                if (ex.Message.StartsWith($"Keyspace {ClusterLockKeyspaceName} does not exist")
+                    || ex.Message.StartsWith($"unconfigured table {ClusterLockTableName}"))
+                    return true;
+                else
+                    return true;
```

Comment:
return true in evrey cases is really the behavior you wanted ?

Reply:
This code is kind of provisional until there is a way to configure Evolve on that matter. If I return false here, it means that having the ```cluster_lock.lock``` table is a hard requirement that cannot be avoided (and is a hard breaking change, although I'm probably the only person in the world using this right now 😄 )

I think it's good practice, but I'd like to include a way to not enforce that using configuration so that you can disable this feature explicitly if you want.

Reply:
Understood. I did not read the related issue ... It's clear now. Thx